src/snowflake/connector/cursor.py

```diff
@@ -1117,8 +1118,25 @@ def fetch_arrow_batches(self) -> Iterator[Table]:
         )
         return self._result_set._fetch_arrow_batches()
 
-    def fetch_arrow_all(self) -> Table | None:
+    @overload
+    def fetch_arrow_all(self, force_return_table: Literal[False] = False) -> None:
+        ...
+
+    @overload
+    def fetch_arrow_all(self, force_return_table: Literal[True] = True) -> Table:
+        ...
+
+    def fetch_arrow_all(self, force_return_table: bool = False) -> Table | None:
+        if not force_return_table:
+            warn(
+                "The behaviour of fetch_arrow_all when the query returns zero rows "
+                "will change from returning None to returning an empty pyarrow table with "
+                "schema using the highest bit length for each column.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
let's add a docstring to the method instead of a warning in run time
```suggestion
        if not force_return_table:
            warn(
                "The behaviour of fetch_arrow_all when the query returns zero rows "
                "will change from returning None to returning an empty pyarrow table with "
                "schema using the highest bit length for each column.",
                DeprecationWarning,
                stacklevel=2,
            )
```